### ISSUE TITLE
SSH Wrapper Automatically Exit after one command

### ISSUE THREAD

**Issue by lyuzizheng** [warpdotdev/Warp#3832]
### Discord username (optional)

brabalawuka

### Operating System

MacOS

### Operating System and Version

macOS 14.1 (23B74)

### Local Shell Version

ZSH 5.9

### Remote Shell Version

Bash 5.2.15(1)-release

### Warp Version

v0.2023.10.31.08.03.stable_02

### I have tried commenting out my system & user rcfiles and warp still will not bootstrap

- [X] Yes

### Screenshots

https://github.com/warpdotdev/Warp/assets/31028570/ec9d2be0-2f17-4936-87df-8e975bba0c8d

### Include shell xtrace output

```
WARP_DEBUG_MODE=1 ssh ec2-user@10.183.10.17
Updates Information Summary: available
    10 Security notice(s)
         9 Important Security notice(s)
         1 Medium Security notice(s)
Run "sudo dnf update" to apply all updates.

A newer release of "Amazon Linux" is available.
  Version 2023.2.20231026:
Run "/usr/bin/dnf check-release-update" for full release and version update info
ls
logstash_start
exit
Connection to 10.183.10.17 closed.
%                                                                             

^[i

```

![CleanShot 2023-11-03 at 11 29 40@2x](https://github.com/warpdotdev/Warp/assets/31028570/da54746d-7c79-4cf2-af7e-8367924d77b9)

It only happens on SSH wrapper.

### Does this block you from using Warp daily?

Yes, this issue prevents me from using Warp daily.

### Is this a Warp specific issue? (i.e. does it happen in Terminal, iTerm, Kitty, etc.)

Yes, this I confirmed this only happens in Warp, not other terminals.

### Warp Internal (ignore) - linear-label:e7dfaa84-5fdb-4a00-b754-d8912da923fa

None

### Warp Internal (ignore): linear-label:b8107fdf-ba31-488d-b103-d271c89cac3e

None

**Comment by dannyneira** [comment]
Hi @Brabalawuka sorry to hear about this blocker. There are some known issues with SSH into amazon instances. You can try adding `ssh` to your `Settings > Subshell` menu, then disable the SSH wrapper by navigating to `Settings > Features` and making sure the default shell on the amazon instance is bash/zsh/fish, then try again.

Please let us know if that helps with your issue.

**Comment by Brabalawuka** [comment]
Yes, without wrapper it works fine but usually an instance from another continent delivers great typing latency. SSH wrapper solves that issue. Hopes that Warp could help improve this!!
